controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/EndpointCertificateMaintainer.java

```diff
@@ -0,0 +1,157 @@
+        this.clock = controller.clock();
+        this.secretStore = controller.secretStore();
+        this.curator = controller().curator();
+        this.endpointCertificateProvider = controller.serviceRegistry().endpointCertificateProvider();
+        this.useEndpointCertificateMaintainer = Flags.USE_ENDPOINT_CERTIFICATE_MAINTAINER.bindTo(controller().flagSource());
+    }
+
+    @Override
+    protected boolean maintain() {
+
+        if (!useEndpointCertificateMaintainer.value())
+            return true; // handled by EndpointCertificateManager for now
+
+        try {
+            // In order of importance
+            deployRefreshedCertificates();
+            updateRefreshedCertificates();
+            deleteUnusedCertificates();
+        } catch (Exception e) {
+            log.log(LogLevel.ERROR, "Exception caught while maintaining endpoint certificates", e);
+            return false;
+        }
+
+        return true;
+    }
+
+    private void updateRefreshedCertificates() {
+        curator.readAllEndpointCertificateMetadata().forEach(((applicationId, endpointCertificateMetadata) -> {
+            // Look for and use refreshed certificate
+            var latestAvailableVersion = latestVersionInSecretStore(endpointCertificateMetadata);
+            if (latestAvailableVersion.isPresent() && latestAvailableVersion.getAsInt() > endpointCertificateMetadata.version()) {
+                var refreshedCertificateMetadata = endpointCertificateMetadata
+                        .withVersion(latestAvailableVersion.getAsInt())
+                        .withLastRefreshed(clock.instant().getEpochSecond());
+                try (Lock lock = lock(applicationId)) {
+                    if (Optional.of(endpointCertificateMetadata).equals(curator.readEndpointCertificateMetadata(applicationId))) {
+                        curator.writeEndpointCertificateMetadata(applicationId, refreshedCertificateMetadata); // Certificate not validated here, but on deploy.
+                    }
+                }
+            }
+        }));
+    }
+
+    /**
+     * If it's been a week since the cert has been refreshed, re-trigger all prod deployment jobs.
+     */
+    private void deployRefreshedCertificates() {
+        var now = clock.instant();
+        curator.readAllEndpointCertificateMetadata().forEach((applicationId, endpointCertificateMetadata) ->
+                endpointCertificateMetadata.lastRefreshed().ifPresent(lastRefreshTime -> {
+                    Instant refreshTime = Instant.ofEpochSecond(lastRefreshTime);
+                    if (now.isAfter(refreshTime.plus(7, ChronoUnit.DAYS))) {
+
+                        controller().jobController().jobs(applicationId).forEach(job ->
+                                controller().jobController().jobStatus(new JobId(applicationId, JobType.fromJobName(job.jobName()))).lastTriggered().ifPresent(run -> {
+                                    if (run.start().isBefore(refreshTime) && job.isProduction()) {
```

Comment:
```suggestion
                                    if (run.start().isBefore(refreshTime) && job.isProduction() && job.isDeployment()) {
```
lest you trigger production tests needlessly.